File: C/logica.c
#include <stdio.h>

int eh_par(int n) {
    return n % 2 == 0;
}

int main() {
    int numero;
    printf("Digite um número: ");
    scanf("%d", &numero);

    if (eh_par(numero)) {
        printf("O número é par.\n");
    } else {
        printf("O número é ímpar.\n");
    }

    printf("Contando até %d:\n", numero);
    for (int i = 1; i <= numero; i++) {
        printf("%d\n", i);
    }

    return 0;
}

File: C#/logica.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Digite um número: ");
        int numero = int.Parse(Console.ReadLine());

        if (EhPar(numero))
        {
            Console.WriteLine("O número é par.");
        }
        else
        {
            Console.WriteLine("O número é ímpar.");
        }

        Console.WriteLine("Contando até " + numero + ":");
        for (int i = 1; i <= numero; i++)
        {
            Console.WriteLine(i);
        }
    }

    static bool EhPar(int n)
    {
        return n % 2 == 0;
    }
}
